subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationNode.java

```diff
@@ -144,6 +139,15 @@ public void resolveDependencies(TaskDependencyResolver dependencyResolver, Actio
             }
         }
 
+        @Override
+        protected ArtifactTransformDependenciesProvider getDependenciesProvider() {
+            if (dependenciesProvider == null) {
+                throw new IllegalStateException("Transformation hasn't been executed yet");
```

Comment:
We should say "initial transformation" here I think. Not that we should ever hit this point...

Reply:
Done.